pkg/util/keyvault/keyvault.go

```diff
@@ -244,32 +245,25 @@ func (m *manager) WaitForCertificateOperation(ctx context.Context, certificateNa
 	return err
 }
 
+type Error struct {
+	AzErr *azkeyvault.Error
+}
+
+func (e *Error) Error() string {
+	return keyvaultError(e.AzErr)
+}
+
+func NewError(err *azkeyvault.Error) error {
+	return &Error{AzErr: err}
+}
+
 func keyvaultError(err *azkeyvault.Error) string {
 	if err == nil {
 		return ""
 	}
 
 	var sb strings.Builder
-
-	if err.Code != nil {
-		sb.WriteString(*err.Code)
-	}
-
-	if err.Message != nil {
-		if sb.Len() > 0 {
-			sb.WriteString(": ")
-		}
-		sb.WriteString(*err.Message)
-	}
-
-	inner := keyvaultError(err.InnerError)
-	if inner != "" {
-		if sb.Len() > 0 {
-			sb.WriteString(": ")
-		}
-		sb.WriteString(inner)
-	}
-
+	spew.Fdump(&sb, err)
```

Comment:
> `spew.Fdump` will generate new lines and indentation. Won't that be escaped in the logs? If so, to make the log more readable would it make sense to use `spew.Fprintf` instead?

Thank you. It might be escaped. Looking at the document of `spew.Fprintf`, it's much better than `Fdump`. I changed to use it.

> Also, have you checked that the whole nested structs and innerErrors are actually printed? IIUC, the spew docs seem to suggest that it will just print the string returned via Error() from the Error interface.

I didn't know that behaviour, but it prints the all attributes because `Error()` is not implemented for `azkeyvault.Error`, so this type is not actually `error`.
I added the test to check innerError to make sure.

Reply:
> `Error()` is not implemented for `azkeyvault.Error`, so this type is not actually `error`
Ah, true, azkeyvault.Error does indeed not implement the `error` interface. Thank you for checking!